independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/InjectionPointInfo.java

```diff
@@ -423,30 +424,44 @@ public boolean equals(Object obj) {
             return true;
         }
 
-        private boolean qualifiersAreEqual(Set<AnnotationInstance> q1, Set<AnnotationInstance> q2) {
-            if (q1 == q2) {
+        private static boolean annotationSetEquals(Set<AnnotationInstance> s1, Set<AnnotationInstance> s2) {
+            if (s1 == s2) {
                 return true;
             }
-            if (q1.size() != q2.size()) {
+            if (s1.size() != s2.size()) {
                 return false;
             }
-            for (AnnotationInstance a1 : q1) {
-                for (AnnotationInstance a2 : q2) {
-                    if (!annotationsAreEqual(a1, a2)) {
+            for (AnnotationInstance a1 : s1) {
+                for (AnnotationInstance a2 : s2) {
+                    if (!annotationEquals(a1, a2)) {
```

Comment:
And why don't we use `AnnotationInstance.equivalentTo(AnnotationInstance)` instead?

Reply:
Because `AnnotationInstance` doesn't have a corresponding hash code function (yet -- see https://github.com/smallrye/jandex/pull/333). Once `AnnotationInstance` has both equivalence and equivalence hash code, we remove both `annotationEquals` and `annotationHashCode`. Until then, I think it's better to keep both `equals` and `hashCode` completely self-sufficient.

In other words, if I switched to `AnnotationInstance.equivalentTo()`, I would be creating another problem of the same kind that I'm fixing here. Potential problem, that is. The `annotationHashCode()` implemented in this commit is consistent with `AnnotationInstance.equivalentTo()`, but there's no stated contract saying that. It would be another dependency on an implementation detail.

Reply:
Well, the contract of `AnnotationInstance#equivalentTo()` is pretty clear so I don't see a dependency on an implementation detail. But let's keep it independent if you think it's better...